CMakeLists.txt

```diff
@@ -1,80 +1,39 @@
 project(git-clang-format-cmake)
 cmake_minimum_required(VERSION 2.8)
 
-set(GCF_GIT_PATH "NOTSET" CACHE STRING "Absolute path to the git executable")
-set(GCF_PYTHON_NAME "python" CACHE STRING "Name of the python executable, eg. python2")
-set(GCF_PYTHON_PATH "NOTSET" CACHE STRING "Absolute path to the python executable")
-set(GCF_CLANGFORMAT_PATH "NOTSET" CACHE STRING "Absolute path to the clang-format executable")
-set(GCF_CLANGFORMAT_STYLE "file" CACHE STRING "Paremter pass to clang-format -style=<here>")
-set(GCF_IGNORE_LIST "" CACHE STRING "Semi colon separeted list of directories to ignore")
+find_package(Git REQUIRED)
+find_package(PythonInterp REQUIRED)
 
-if("${GCF_GIT_PATH}" STREQUAL "NOTSET")
-	find_program(FIND_GIT git)
-	if("${FIND_GIT}" STREQUAL "FIND_GIT-NOTFOUND")
-		message(FATAL_ERROR "Could not find 'git' please set GCF_GIT_PATH:STRING")
-	else()
-		set(GCF_GIT_PATH ${FIND_GIT})
-		message(STATUS "Found: ${GCF_GIT_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_GIT_PATH})
-		message(WARNING "Could not find git: ${GCF_GIT_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_GIT_PATH}")
-	endif()
-endif()
-
-if("${GCF_PYTHON_PATH}" STREQUAL "NOTSET")
-	find_program(FIND_PYTHON ${GCF_PYTHON_NAME})
-	if("${FIND_PYTHON}" STREQUAL "FIND_PYTHON-NOTFOUND")
-		message(FATAL_ERROR "Could not find '${GCF_PYTHON_NAME}' please set GCF_PYTHON_PATH:STRING")
-	else()
-		set(GCF_PYTHON_PATH ${FIND_PYTHON})
-		message(STATUS "Found: ${GCF_PYTHON_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_PYTHON_PATH})
-		message(WARNING "Could not find python: ${GCF_PYTHON_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_PYTHON_PATH}")
-	endif()
-endif()
-
-if("${GCF_CLANGFORMAT_PATH}" STREQUAL "NOTSET")
-	find_program(FIND_CLANGFORMAT clang-format)
-	if("${FIND_CLANGFORMAT}" STREQUAL "FIND_CLANGFORMAT-NOTFOUND")
-		message(FATAL_ERROR "Could not find 'clang-format' please set GCF_CLANGFORMAT_PATH:STRING")
-	else()
-		set(GCF_CLANGFORMAT_PATH ${FIND_CLANGFORMAT})
-		message(STATUS "Found: ${GCF_CLANGFORMAT_PATH}")
-	endif()
-else()
-	if(NOT EXISTS ${GCF_CLANGFORMAT_PATH})
-		message(WARNING "Could not find clang-format: ${GCF_CLANGFORMAT_PATH}")
-	else()
-		message(STATUS "Found: ${GCF_CLANGFORMAT_PATH}")
-	endif()
-endif()
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})
+find_package(ClangFormat REQUIRED)
```

Comment:
If `FindClangFormat.cmake` reporets the `clang-format` version then the addition of a `GCF_CLANGFORMAT_VERSION` option would make a nice addition here, so the including project could require a specific version.